pkgs/development/compilers/oraclejdk/jdk17-linux.nix

```diff
@@ -0,0 +1,55 @@
+{ lib, stdenv
+, requireFile
+, xorg
+, zlib
+, freetype
+, alsa-lib
+, setJavaClassPath
+, fetchzip
+}:
+
+let result = stdenv.mkDerivation rec {
+  pname = "oraclejdk";
+  version = "17.0.3.1";
+
+  src = fetchzip {
+    name = "jdk-${version}_linux-x64_bin.tar.gz";
+    url = "https://www.oracle.com/java/technologies/downloads/#java17";
```

Comment:
With this URL I get the following error:

```builder for '/nix/store/3iixmm7982ry728cjigxv5hdkqwwdkw9-jdk-17.0.3.1_linux-x64_bin.tar.gz.drv' failed with exit code 1;
       last 10 log lines:
       >
       > trying https://www.oracle.com/java/technologies/downloads/#java17
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       > 100  148k    0  148k    0     0   951k      0 --:--:-- --:--:-- --:--:--  958k
       > unpacking source archive /tmp/nix-build-jdk-17.0.3.1_linux-x64_bin.tar.gz.drv-0/#java17
       > tar: This does not look like a tar archive
       > tar: Skipping to next header
       > tar: Exiting with failure status due to previous errors
       > do not know how to unpack source archive /tmp/nix-build-jdk-17.0.3.1_linux-x64_bin.tar.gz.drv-0/#java17
       For full logs, run 'nix log /nix/store/3iixmm7982ry728cjigxv5hdkqwwdkw9-jdk-17.0.3.1_linux-x64_bin.tar.gz.drv'.
```

Reply:
Ah, sure, forget to change `fetchzip` to `requireFile`, fixed. However not sure if the direct link to tar can be added here since I'm not sure about the license.